src/hooks/useENSRegistrationStepHandler.tsx

```diff
@@ -112,7 +123,25 @@ export default function useENSRegistrationStepHandler(observer = true) {
         startPollingWatchCommitTransaction();
       }, 2000);
     }
-  }, [registrationStep, watchCommitTransaction]);
+  }, [observer, registrationStep, watchCommitTransaction]);
+
+  useEffect(() => {
+    // we need to update the loop with new commit transaction hash in case of speed ups
+    if (
+      !observer &&
+      !!prevCommitTrasactionHash &&
+      !!commitTransactionHash &&
+      prevCommitTrasactionHash !== commitTransactionHash
+    ) {
+      timeout.current && clearTimeout(timeout.current);
+      startPollingWatchCommitTransaction();
```

Comment:
if the `commitTransactionHash` in case of speed ups we should start the process again because this dep change doesn't update inside timeouts